### PR TITLE
Added option to set message log width fullscreen or sidebar visible

### LINKED ISSUE #46172: Bigger or customizable size log window
# Is your feature request related to a problem? Please describe.
Currently, message log takes about 1/4th of the screen. For comparison, options window takes about 1.5-2 times more. If I'm opening the log it should be easy to read, and currently... well, I wouldn't say this is not the case but it could be better.

# Describe the solution you'd like
Make log window at least same size as options window. Or make the size adjustable (through options-interface, I think?)

# Describe alternatives you've considered
Continue using current tiny window which isn't much of an upgrade compared to sidebar log panel.

# Additional context
cataclysm-tiles_2020_12_20_12_31_31_868

### PR DESCRIPTION
**Pull request by Night-Pryanik**
#### Summary

SUMMARY: Interface "Added option to set message log width fullscreen or sidebar visible."

#### Purpose of change
Make message log utilize modern monitors resolution.
Closes #46172.

#### Describe the solution
Added an option to set message log width fullscreen or sidebar visible.

#### Describe alternatives you've considered
None.

#### Testing
Opened message log ingame. Toggled option to fullscreen or sidebar visible. Changed game width to check how it affects message log width.

#### Additional context
Before:
изображение

After, sidebar visible:
изображение

### CODE REVIEW

**Comment by anothersimulacrum** [comment]
I am not sure this needs an option, sidebar visible seems like it would work for everyone.

**Comment by Night-Pryanik** [comment]
I'm not attached to fullscreen option, I can remove it if it seems redundant.

File: src/options.cpp [hunk 1]
```diff
@@ -1597,6 +1597,11 @@ void options_manager::add_options_interface()
          false
        );
 
+    add( "MESSAGE_LOG_WIDTH", "interface", to_translation( "Full screen message log" ),
+         to_translation( "If true, message log will fit full screen, otherwise it will leave sidebar visible." ),
+         false
+       );
+
```
Comment (kevingranade) [inline src/options.cpp:1604]:
```suggestion
```

File: src/messages.cpp [hunk 2]
```diff
@@ -472,8 +473,10 @@ Messages::dialog::dialog()
 
 void Messages::dialog::init( ui_adaptor &ui )
 {
-    w_width = std::min( TERMX, FULL_SCREEN_WIDTH );
-    w_height = std::min( TERMY, FULL_SCREEN_HEIGHT );
+    w_width = get_option<bool>( "MESSAGE_LOG_WIDTH" ) ? TERMX :
+              std::max( 45, TERMX - 2 * ( panel_manager::get_manager().get_width_right() +
+                                          panel_manager::get_manager().get_width_left() ) );
```
Comment (kevingranade) [inline src/messages.cpp:478]:
```suggestion
    w_width = std::max( 45, TERMX - 2 * ( panel_manager::get_manager().get_width_right() +
                                          panel_manager::get_manager().get_width_left() ) );
```

**Pull request merged**